### DISCUSSION TITLE
Android Optimization Support

### DISCUSSION THREAD

**Discussion by joeyoropesa-dev** [tsukasa/BdBrowser#22] [Ideas]
After kiwi browser project, android users can use desktop extension in that chromium-based browser without any issues. Can you maybe add an option in this extension called Mobile Integration to modify BetterDiscord UI in the browser to look like an Android app so that visuals, buttons and etc can be useful on small touchscreens 🥰

Every other features can be untouched because I tested it on Android phone and it works perfectly 🫂

Thank you so much for this beautiful wonderful forked project.

**Comment by Serega007RU** [comment]
There is a modified discord on android, for example Aliucord
Why would you do this through a browser on your phone? This is too weird

**Reply by ClaraCrazy** [reply]
if only bluecord would still be maintained. it was (IMO) a lot cleaner. (Lets not discuss the scummy developer)

but fr. Running this through browser on mobile seems weird

**Reply by joeyoropesa-dev** [reply]
Not with support of the latest Discord version due to the react native integration by Discord that is trying to kill modding their product on mobile devices (to make it hard to modify it) so almost all of Aliucord plugins are now EOL - some of them are ported but most of them are literally dead forever. BetterDiscord is still supported and have a very big number of plugins and themes - even favorite ones that doesn't exist for Aliucord.

Sorry for letting you to know this but this is the fact - the truth.

**Reply by ClaraCrazy** [reply]
I still run bluecord lol. I dont give a damn about any of the new, bloated, useless and distracting discord crap they add. Luckily they are smart (or stupid depending on who you ask) and make most of the important stuff 100% web-based, and it just embeds nicely into the app without an update. All im missing as of now is super reactions & the new dm request thingy, neither of which I use.

As for aliu... never used it but I hope you realize how Open-source works. Just because discord pushed an update and things dont work right this second doesnt mean they are "dead forever", thats just not how it works. I'd know, I'm a developer myself working on lots of things. Discord is a huge company with roughly 1k employees. Those plugins are made by one person each. The react change was in fact not to "kill modding" either. It was a loooong needed change that, from a code perspective, was the only way forward. Anyways... patience is a virtue.

Running discord in browser on mobile will never work properly. Its not mobile-optimized, things work entirely differently, mobile browsers lack features desktop browsers have, layouts for mods would need to be manually adjusted for each single mod that even adds a single line of new css or modifies one... You are asking for a nightmare when you have two fully functional solutions

**Reply by joeyoropesa-dev** [reply]
> I still run bluecord lol. I dont give a damn about any of the new, bloated, useless and distracting discord crap they add. Luckily they are smart (or stupid depending on who you ask) and make most of the important stuff 100% web-based, and it just embeds nicely into the app without an update. All im missing as of now is super reactions & the new dm request thingy, neither of which I use.
> 
> As for aliu... never used it but I hope you realize how Open-source works. Just because discord pushed an update and things dont work right this second doesnt mean they are "dead forever", thats just not how it works. I'd know, I'm a developer myself working on lots of things. Discord is a huge company with roughly 1k employees. Those plugins are made by one person each. Patience is a virtue.
> 
> Running discord in browser on mobile will never work properly. Its not mobile-optimized, things work entirely differently, mobile browsers lack features desktop browsers have, layouts for mods would need to be manually adjusted for each single mod that even adds a single line of new css or modifies one... You are asking for a nightmare when you have two fully functional solutions

Thank you so much for your side of thinking about this idea but because I've got some attention by this fork chrome extension developer, I would love to not give him a reason to cancel this idea because, at least for most people like me it would be nice to have so much plugins, better support for them, new Discord features because most of the servers is moving their legacy features from the bot to the new Discord built-in features in the new Discord versions so.... even after all, it can't be so much bad to have this feature - at least as an optional feature so that the person who wants to experience this can enable that feature.


Anyway, getting a BetterDiscord theme that will modify the look and feel of Discord to be more mobile-friendly will be even that enough instead of nothing.

Thank you for your understanding!

**Comment by tsukasa** [comment]
Greetings,

thank you for your suggestion and your kind words! Setting aside the question of native client versus web client on mobile devices, theming the Discord client should be done through a client mod's existing theme system. I am open to discussing options for overriding/setting themes out of the box (i.e. by overwriting the contents of BetterDiscord's custom.css with a local file or something along those lines).

However, building and maintaining a complete Discord theme that turns the web client into a more mobile-friendly experience is not on my to-do list due to the required amount of work, the maintenance cost this would bring, and the fact I would not be using the theme myself (which would probably make this a "second-class citizen" right from the start).

**Reply by joeyoropesa-dev** [reply]
It's ok - i don't ask how much time we need to wait for you to finish. Do it when you're free even if with that speed we need to wait for years to get mobile-friendly UI and etc...

I'm just happy that you're much better than BetterDiscord team that really doesn't even care about adding support for mobile devices but you - you have a talent and resources and everything you need to make it to happen

But I don't hurry you - take your time you need for this - anyway, it's not easy at all to implement something that is build it only for desktop to mobile devices.

We love you our best dev and savior of the Discord modding feature on those devices 🥰🥰🥰

Of course, you can make donations for this high type of support

The standard mobile support that will be free completely can be the thing you can do it without so much costs and etc...

So I agree with you ✨✨✨
